src/components/autocomplete/js/autocompleteDirective.js

```diff
@@ -325,6 +328,19 @@ function MdAutocomplete ($$mdSvgRegistry) {
       // will hold the actual tabindex.
       element.attr('tabindex', '-1');
 
+      var disableVirtualRepeat = $mdAutocompleteCtrl.scope.disableVirtualRepeat;
+      var containerType = disableVirtualRepeat ? 'div' : 'md-virtual-repeat-container';
+      var repeatType = disableVirtualRepeat ? 'ng-repeat' : 'md-virtual-repeat';
+      var intermediateTags = disableVirtualRepeat ?
+            '\
+            <div class="md-autocomplete-variable-height-list-container md-whiteframe-z1" role="presentation"\
+                ng-class="{ \'ng-hide\': $mdAutocompleteCtrl.hidden }">\
```

Comment:
It looks like `blur()` on the controller is setting `hasFocus = false` too early. It should set it after the block where it updates `ctrl.hidden`. By setting it early, it's causing `shouldHide()` to return `true` on blur, which it should not. It should wait until later in the event handling when it handles the `ng-click` and calls `select()` on the controller. 

I tested moving it after that block and it seems to work w/o side effects.

Reply:
I think that we should drop the approach of trying to dynamically place the tag (i.e. `<' + containerType + '\`). This approach ends up with `md-auto-shrink` and `md-auto-shrink-min="1"` on a `div` in the case where `disable-virtual-repeat` is set. This doesn't make sense since they are `md-virtual-repeat-container` directives and shouldn't be on ` div`.

We probably need a `getRepeatContainer()` function that returns the proper partial template. We may also need a `getClosingTags()` function.

Reply:
I like it. I'll go ahead and refactor this.

Reply:
What do you think of this?

Reply:
What do you think of the revision?

Reply:
I was waiting for some of the rebase and merges to get cleaned up. Taking a deeper look now...